Tasks/e2_dijkstra.py

```diff
@@ -3,11 +3,24 @@
 
 
 def dijkstra_algo(g: nx.DiGraph, starting_node: Any) -> dict:
-	"""
-	Count shortest paths from starting node to all nodes of graph g
-	:param g: Graph from NetworkX
-	:param starting_node: starting node from g
-	:return: dict like {'node1': 0, 'node2': 10, '3': 33, ...} with path costs, where nodes are nodes from g
-	"""
-	print(g, starting_node)
-	return dict()
+    """
+    Count shortest paths from starting node to all nodes of graph g
+    :param g: Graph from NetworkX
+    :param starting_node: starting node from g
+    :return: dict like {'node1': 0, 'node2': 10, '3': 33, ...} with path costs, where nodes are nodes from g
+    """
+    visited_nodes = []
+    cost_dict = {node: float('inf') if node != starting_node else 0 for node in g.nodes}
+
+    while len(visited_nodes) < len(g.nodes):
+
+        min_node = min(filter(lambda key: key not in visited_nodes, cost_dict), key=lambda key: cost_dict[key])
```

Comment:
🥇 
Неблокирующий

Reply:
Лучше, правда было бы сделать что-то вроде:
```suggestion
        min_node = min(set(cost_dict.keys()) - visited_nodes), key=lambda key: cost_dict[key])
```
При условии, что visited_nodes - множество (а нет причин ему НЕ быть множеством)
Неблокирующий